service/arm_templates/main.json

```diff
@@ -50,28 +50,93 @@
     "kubernetesIngressInsecurePort": {
       "type": "int"
     },
+    "masterCount": {
+      "type": "int",
+      "defaultValue": 1
+    },
+    "masterIPAddresses": {
+      "type": "array",
+      "metadata": {
+        "description": "An array containing all the master nodes' ip addresses."
+      }
+    },
+    "masterVMSize": {
+      "type": "string"
+    },
+    "masterDataDiskSizeGB": {
+      "type": "int"
+    },
+    "masterAdminUsername": {
+      "type": "string"
+    },
+    "masterAdminPassword": {
+      "type": "string"
+    },
+    "masterDisablePasswordAuth": {
+      "type": "bool",
+      "defaultValue": true
+    },
+    "masterAdminSSHKeyPath": {
+      "type": "string"
+    },
+    "masterAdminSSHKeyData": {
+      "type": "string"
+    },
+    "masterOSImagePublisher": {
+      "type": "string"
+    },
+    "masterOSImageOffer": {
```

Comment:
What is the difference between Publisher and Offer? We'll need to add the values for the CoreOS image to `azuretpr`.

Reply:
If GiantSwarm built their own CoreOS image, we could have Publisher:GiantSwarm Offer:CoreOS

Reply:
This is how it's defined on Azure:

```
 +          "imageReference": {
 +            "publisher": "[parameters('masterOSImagePublisher')]",
 +            "offer": "[parameters('masterOSImageOffer')]",
 +            "sku": "[parameters('masterOSImageSKU')]",
 +            "version": "[parameters('masterOSImageVersion')]"
 +          },
```